### DISCUSSION TITLE
UI Restyling

### DISCUSSION THREAD

**Discussion by adrianptr** [elementary/stylesheet#901] [Ideas]
Premise: mine is to be taken only as a daily user advice.

I think it is appropriate to make Elementary OS graphically more modern: windows, icons, animations, notification area, buttons and widgets. It could give this awesome OS some extra momentum.

I'm sorry for my poor english.

**Comment by cassidyjames** [comment]
This is so broad. elementary is constantly iterating on our visual style, icons, animations, etc. elementary OS 6 will be bringing our latest updates there, plus a new stylesheet that feels similar to the old one while having much more consistent lighting/shadow/contrast/color usage.

In general, elementary does not chase design trends; just saying "elementary OS should be more modern" is not really actionable, and comes off as, "elementary OS should chase more trends." :stuck_out_tongue:

I've left this as a discussion for now in case there are any specific examples that could be raised, but as-is, this is not actionable at all.

**Comment by danirabbit** [comment]
Yeah if you could elaborate on what you mean by "graphically more modern" maybe we could generate some discussion around certain elements and how they could be made nicer

**Reply by adrianptr** [reply]
Thanks everyone for the answers. Analyzing the desktop environment, they could be improved:

- Window shapes, window shadows and buttons (minimize, close, etc)
- Make toolbar buttons and tabs more pleasing to the eye (e.g. in the Files app)
- Application menu structure
- Notification area and popups (Shape and shadows)
- More vivid colors
- Simpler icons (less particular and more colorful)
- Change the icon of workspaces and desktop (they are not very intuitive)

For the Files app I also wanted to propose to insert more features regarding the sorting of files and advanced search functions.

**Reply by Chandram-Dutta** [reply]
> 
> - Window shapes, window shadows and buttons (minimize, close, etc)

Ig they are quite suitable now... And gives elementary it's unique quality. 

> - Make toolbar buttons and tabs more pleasing to the eye (e.g. in the Files app)

The toolbar buttons are quite good. In eOS6 the tabs design has been improved. 

> - Application menu structure

Umm... The present Application Menu Structure is quite good. However open to more intuitive option

> - Notification area and popups (Shape and shadows)

With eOS6, this has been improved. 

> - More vivid colors

In eOS6, the introduction of accent colours really stands out across the shell

> - Simpler icons (less particular and more colorful)

The icons(application) go with the elementary HIG and hence with rest of the theme. It helps eOS and app developers to nation uniformity. 

> - Change the icon of workspaces and desktop (they are not very intuitive)

They are good...However new ideas could be discussed. 

> For the Files app I also wanted to propose to insert more features regarding the sorting of files and advanced search functions.

You can open PRs for the functions and features.

**Reply by janopae** [reply]
Thank you for being so open regarding this discussion. I really like the fact that elementary OS is not running behind every new design trend. This gives the design a chance to mature and become better and more functional over time. However, by the time of writing, the design looks extremely dated. So dated, that I'd hesitate installing it on a new Dell XPS laptop, because the laptop looks so stylish and new while the OS looks so dated.

I actually had some ideas for improvements floating in my had before I found this discussion.

## Hover effects

I think, the elementary OS design is lacking hover effects. This is not just an UI trend, but actually something that is usable in terms of design being able to be "objectively good": Hover effects give a final assurance about something being clickable. Clickability should be indicated without hovering, but sometimes, this indication is not that clear. Icon Buttons in the headerbar don't look like buttons in elementary OS, and adding a hover effect would finally make them "feel" clickable. A Mac user might not know that clicking the time opens a really useful calendar. In Gnome, you have a nice hover effect indicating this. In the elementary calendar app, I was at first not sure if clicking dates had an effect. It has.

Even if we only thought about the aesthetical aspect of hover effects, we have to accept that they are common in operating systems since about 20 years now. Even if we expect people to get rid of their old habits and fully endorse the "elementary style", they won't escape hover effects, as they are part of most web designs these days – including elementary's own website. That's why I say, elementary OS is lacking these.

Header bar items have a circle drawn behind them when clicked. I propose drawing a lighter version of that circle when hovering. Buttons might subtly darken their backround colour.

## Deciding on a style for header bar icons

Currently, elementary OS is mixing monochrome and coloured icons in the header bars: While Files has only monochrome icons in its header bar, Mail and Calendar have colourful ones. Because window controls are always monochrome, they mix colourful icons with monochrome icons. Especially the settings wheel to me is an object of personal annoyance, because it is usually placed directly next to the monochrome maximize button.

I think, colourful icons and monochrome icons should be used consistently. I propose using monochrome icons only in header bars. Maybe we should be following Gnome HIG on this.

## Getting rid of the bottom drop shadow in icons

This is just a non-objective thing about aesthetics, however, I would not describe it as "running behind a UI trend". I'd rather describe it as "giving up an UI trend that has been ran behind when the current icons were created". You see that little, round shadow at the bottom especially of header bar icons, that gives them a look of sort of ortogonally standing on top of the laid-back header bar? I really think it should go.

## Triangle style UI elements

Some rare UI elements, like the current path in Files, have a pike, that indicates a direction. This looks kind of out of place within the rest of elementary's design, which is much more rounded. These pikes should be rounded, too, or replaced with a diffrerent design (like a single triangle or arrow between the elements, like Nautilus or Dolphin do it).

## Shadows in sidebar-bottom buttons

Some elementary apps have buttons at the very bottom of their sidebar (e. g. Files, To do, Code). These buttons look like they were carved into the sidebar, with shadows indicating they were on a lower level. I argue that this design makes no sense and that these buttons either should look like normal buttons, or like normal sidebar elements. They should be separated with nothing more than a line.

## Make selected elements in sidebars consistently use the highlight colour

Selected elements in the sidebar of the system settings use the highlight colour (blue by default) to indicate they're selected. Files and others use grey instead. I suggest using the highlight colour, to make the user interface more colourful.

## Make shadows and gradiants more subtle in general (maybe increase the contrast between background colours instead)

This is just a personal preference, as well as a current UI trend, but I think gradients and shadows through the UI should be a bit less extreme and more subtle. Also, the colours could be brighter in light mode. I think, GitHub is a good example of using gradiants, as well as shadows and even highlights, but using them subtly enough to still look modern. If elementary OS looked a little bit more like GitHub, I would be happy.

**Reply by Chandram-Dutta** [reply]
> However, by the time of writing, the design looks extremely dated.

Yes. I agree. However ig it isn't about the design style. Its more about the colour choice. Especially the light theme. The dark theme really looks modern and up to date.


> ## Hover effects

Yes, a good topic to discuss more about. 

> ## Deciding on a style for header bar icons

About this topic. I believe the elementary HIG should be more direct about what style to choose for HeaderBar icons and the header bar itself(flat or the legacy one) like Task follow the flat design however setting and all others don't. 

 #### I really can't comment on other topics coz I personally never felt any problem with those...but will check into those.

**Reply by Chandram-Dutta** [reply]
> However, by the time of writing, the design looks extremely dated.

Yes. I agree. However ig it isn't about the design style. Its more about the colour choice. Especially the light theme. The dark theme really looks modern and up to date.

> ## Hover effects

Yes, a good topic to discuss more about. 

> ## Deciding on a style for header bar icons

 About this topic. I believe the elementary HIG should be more direct about what style to choose for HeaderBar icons and the header bar itself(flat or the legacy one) like Task follow the flat design however setting and all others don't. 

#### I really can't comment on other topics coz I personally never felt any problem with those...but will check into those.
